app/models/base_facility.rb

```diff
@@ -201,6 +204,20 @@ def build_distance_result_set(lat, long, type, services, ids, limit = nil)
     end
     # rubocop:enable Metrics/ParameterLists
 
+    def zip_query(params)
+      # TODO: allow user to set distance from zip
+      zip_plus0 = params[:zip][0...5]
+      zcta = CSV.read(Rails.root.join('modules', 'va_facilities', '2018_Gaz_zcta_national.txt'), col_sep: "\t")
+      requested_zip = zcta.select { |area| area[0] == zip_plus0 }
+      requested_zip = [['x']] unless requested_zip.any?
```

Comment:
Why does it default to `'x'` in this case? Does that return everything or nothing when it queries the jsonb?

Reply:
`x` causes it to return no results (because no zip codes start with an `x`). Which I figure is better than returning everything if the zip code you request doesn't exist.

Reply:
Can you skip the query altogether and return no results?

Reply:
Actually, can we move this up to the controller level and send a 4xx status code back if it fails this  validation?

Reply:
>Can you skip the query altogether and return no results?

This would be easier to read and faster. :+1: 

>Actually, can we move this up to the controller level and send a 4xx status code back if it fails this validation? There's a Common::Exceptions::InvalidFieldValue that you could use.

Yeah, as I was wrapping things up I started thinking about validating values. We could return an error code in these cases. Besides a patently invalid input like `taco` we could also end up with a superficially valid string in the format `#####` or `#####-####` that doesn't actually correspond to an actual zip code. If we want to return an error code during validation for those (as opposed to simply returning an empty array) I think it will require another look up in the ZCTA array to suss out.